packages/fluence-js/src/internal/FluencePeer.ts

```diff
@@ -690,6 +690,11 @@ export class FluencePeer {
                 // execute call requests if needed
                 // and put particle with the results back to queue
                 if (item.result.callRequests.length > 0) {
+                    if (item.particle.hasExpired()) {
```

Comment:
Maybe do this check at the very beginning of the subscribe callback? You surely don't want to send expired particles to the network. 

And call something like `onStageChange('expired')`.

Reply:
Basically you would do `filterExpiredParticles` after interpretation. I do expiration checks on rust peer after interpretation as well, too.

Reply:
Yeah, makes sense